pkg/gameserverallocations/allocation_cache.go

```diff
@@ -231,13 +232,83 @@ func (c *AllocationCache) ListSortedGameServers() []*agonesv1.GameServer {
 			}
 		}
 
+		// TODO: if we end up here, then break the tie with count/list preference.
+
 		// finally sort lexicographically, so we have a stable order
 		return gs1.Status.NodeName < gs2.Status.NodeName
 	})
 
 	return list
 }
 
+// PrioritySortGameServers returns a list of the cached gameservers sorted by Priorities.
+func (c *AllocationCache) PrioritySortGameServers(gsa *allocationv1.GameServerAllocation) []*agonesv1.GameServer {
+	list := c.getGameServers()
+	if list == nil {
+		return []*agonesv1.GameServer{}
+	}
+
+	sort.Slice(list, func(i, j int) bool {
+		gs1 := list[i]
+		gs2 := list[j]
+
+		for _, priority := range gsa.Spec.Priorities {
```

Comment:
Looks like we are literally only sorting GameServers according to the the first `Priority` of the `gsa.Spec.Priorities` list? So why the input of Allocation has a list of `Priority`? Do we want customer to specify a list of `Priority` that should be considered during allocation? If so, should we treated the list weighted? i.e. sorting GameServers based on the Priority at the beginning of the list, and only looking at the subsequent Priorities to break tie?

Reply:
The first Priority in the list is considered, and only in the case of a tie is a subsequent priority(s) considered. I'm currently pulling out this logic into a separate comparator.

Reply:
SG. We should explicitly called out in doc and API reference that the Priorities are weighted.

Reply:
I like the trick - if `compareGameServers` returns 0, then it passes through to the next one. I like it 👍🏻